app/scripts/services/homePagePreference.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .service("HomePagePreference", function($location, $timeout, $q, $uibModal, AuthService, Logger, Navigate, NotificationsService){
+
+    var HOME_PAGE_KEY_PREFIX = "openshift/home-page-pref/";
+
+    function getHomePagePreferenceKey() {
+      return HOME_PAGE_KEY_PREFIX;
+    }
+
+    var getHomePagePreference = function() {
+      var homePagePref;
+      var key = getHomePagePreferenceKey();
+
+      if (!key) {
+        return 'catalog-home';
+      }
+
+      try {
+        homePagePref = JSON.parse(localStorage.getItem(key));
+      } catch (e) {
+        Logger.error('Could not parse homePagePref as JSON');
+        return 'catalog-home';
+      }
+
+      return _.get(homePagePref, 'type', 'catalog-home');
+    };
+
+    var getProjectOverviewPage = function() {
```

Comment:
nit: It's getting a project name, not an overview page (which to me implies a URL). Maybe

```js
var getHomePageProjectName = function() ...
```